comp/core/secrets/secretsimpl/rotating_ndrecords_test.go

```diff
@@ -0,0 +1,395 @@
+{"time":"2014-02-04T10:30:10Z","data":[{"handle":"banana"}]}
+{"time":"2014-02-04T10:30:20Z","data":[{"handle":"cherry"}]}
+`)
+	os.WriteFile(tmpFileName, startingData, 0640)
+
+	// create with a size limit would be reached by the next row
+	records := newRotatingNDRecords(tmpFileName, config{sizeLimit: 150, retention: 35 * time.Second})
+	mockClock.Add(30 * time.Second)
+	records.Add(mockClock.Now(), []auditRecord{{Handle: "donut"}})
+
+	// there is 1 rotated file, set its mtime so it will be removed soon
+	rotated := records.RotatedFiles()
+	assert.Equal(t, 1, len(rotated))
+	os.Chtimes(rotated[0], time.Now(), mockClock.Now())
+
+	// only 1 record in the current file, the latest
+	data, _ := os.ReadFile(tmpFileName)
+	expect := `{"time":"2014-02-04T10:30:30Z","data":[{"handle":"donut"}]}
+`
+	assert.Equal(t, expect, string(data))
+
+	// advance time and add another record
+	mockClock.Add(20 * time.Second)
+	records.Add(mockClock.Now(), []auditRecord{{Handle: "egg"}})
+
+	// there are 2 records in the audit file
+	data, _ = os.ReadFile(tmpFileName)
+	expect = `{"time":"2014-02-04T10:30:30Z","data":[{"handle":"donut"}]}
+{"time":"2014-02-04T10:30:50Z","data":[{"handle":"egg"}]}
+`
+	assert.Equal(t, expect, string(data))
+
+	// there is 1 rotated file still
+	rotated = records.RotatedFiles()
+	assert.Equal(t, 1, len(rotated))
+
+	// advance time again and add another record
+	mockClock.Add(20 * time.Second)
+	records.Add(mockClock.Now(), []auditRecord{{Handle: "fruit"}})
+
+	// there are 0 rotated files because the last append removed the old rotated file
+	rotated = records.RotatedFiles()
+	assert.Equal(t, 0, len(rotated))
+}
+
+// Test the regex and fmt pattern that gets built from config
+func TestBuildRotationRegexAndName(t *testing.T) {
+	type testCase struct {
+		description  string
+		filename     string
+		spacer       int
+		num          int
+		expectRegex  string
+		expectFmtPat string
+	}
+
+	testCases := []testCase{
+		{
+			description:  "build regexp with spacer 4",
+			filename:     "/tmp/file.txt",
+			spacer:       4,
+			num:          7,
+			expectRegex:  `file\.(\d{4})\.txt`,
+			expectFmtPat: "/tmp/file.0007.txt",
+		},
+		{
+			description:  "build regexp with spacer 8",
+			filename:     "/tmp/file.txt",
+			spacer:       8,
+			num:          19,
+			expectRegex:  `file\.(\d{8})\.txt`,
+			expectFmtPat: "/tmp/file.00000019.txt",
+		},
+		{
+			description:  "build regexp with no file extension",
+			filename:     "/tmp/file",
+			spacer:       4,
+			num:          121,
+			expectRegex:  `file\.(\d{4})`,
+			expectFmtPat: "/tmp/file.0121",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			re, err := buildRotationRegex(tc.filename, tc.spacer)
+			if err != nil {
+				t.Error(err)
+			}
+			assert.Equal(t, tc.expectRegex, re.String())
+			rotName, err := buildRotationName(tc.filename, tc.spacer, tc.num)
+			assert.NoError(t, err)
+			assert.Equal(t, tc.expectFmtPat, rotName)
+		})
```

Comment:
Would it be valuable to double check that the generated filename matches the generated regex ?

```suggestion
			assert.True(t, re.MatchString(rotName))
		})
```